test/components/form-draft/publish.spec.js

```diff
@@ -269,11 +269,51 @@ describe('FormDraftPublish', () => {
       .afterResponse(modal => {
         modal.should.alert(
           'danger',
-          'The version name you’ve specified conflicts with a past version of this Form. Please change it to something new and try again.'
+          'The version name of this Draft conflicts with a past version of this Form or a deleted Form. Please use the field below to change it to something new or upload a new Form definition.'
         );
       });
   });
 
+  it('shows the version input field after request returns duplicate version problem', () => {
+    // The scenario here is a user trying to publish a form that conflicts with
+    // a form/version combo probably found in the trash. This component doesn't
+    // have access to trashed forms so it doesn't know about the problem until the
+    // request from the backend returns the problem.
```

Comment:
I think this comment is helpful, and I wonder whether it makes sense to move it to the component itself so that the motivation behind the approach there is clear. Just my two cents, it also works well here!

Reply:
Now included in the component, too!